lib/TelemetrySender.js

```diff
@@ -1,21 +1,45 @@
 /* globals Services */
 
 const {Cu} = require("chrome");
+const self = require("sdk/self");
 Cu.import("resource://gre/modules/Services.jsm");
 Cu.importGlobalProperties(["fetch"]);
 
+// The preference that tells what locale the user selected
+const PREF_SELECTED_LOCALE = "general.useragent.locale";
+
 function TelemetrySender(pingEndpoint) {
   Services.obs.addObserver(this, "tab-session-complete");
   this._pingEndpoint = pingEndpoint;
 }
 
 TelemetrySender.prototype = {
+  get locale() {
+    try {
+      return Services.prefs.getCharPref(PREF_SELECTED_LOCALE);
+    }
+    catch (e) {
+      // User has no locale set. We will default to en-US.
+    }
+    return "en-US";
+  },
+
   observe: function(subject, topic, data) {
     if (topic === "tab-session-complete") {
+      data = this._appendRequiredPingData(data);
       this._sendPing(data);
     }
   },
 
+  _appendRequiredPingData: function(data) {
+    let ping = JSON.parse(data);
+    ping.addon_version = self.version;
```

Comment:
since this is unlikely to change, i suggest we move all of this data to TabTracker instead of requiring a deserialize/serialize